### ISSUE TITLE
Error creating new RADON workspace

### ISSUE THREAD

**Issue by sdagostini** [radon-h2020/radon-ide#24]
If an user try to create/open a RADON workspace and another RADON workspace has been already opened by another user, the following error occurs:

Error Failed to run the workspace Failure executing POST at https10.96.0.1apiv1namespacescheservices. Message services radon-gmt already exists. Received status Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=services, name=radon-gmt, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=services radon-gmt already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={}).

**Comment by miwurster** [comment]
Hi @sdagostini, I also experienced this error in my environment. This error is raised because old workspaces didn't clean up every Kubernetes resource on the underlying cluster --> I needed to delete the `radon-gmt` service by myself.

The following `devfile` config creates this Kubernetes service: 

```
    endpoints:
      - name: radon-gmt
        port: 8080
        attributes:
          protocol: http
          secure: "false"
          discoverable: "true"
          public: "true"
```

IMHO this is a bug in Eclipse Che. WDYT?

**Comment by sdagostini** [comment]
Hi @miwurster, I experimented this error when two different user are using at the same time a RADON workspace. At the moment it sems that 2 different users can not simultaneously use 2 different workspaces. I think that this error could be related to the configuration on the winery.yaml file:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: winery-deployment
  labels:
    app: winery
    tier: frontend
spec:
  replicas: 1
  selector:
    matchLabels:
      app: winery
      tier: frontend
...
```

The "replicas" field is set to 1, maybe we have to set more replicas in order to make available more replicas of the GMT to the users (see https://kubernetes.io/docs/concepts/workloads/controllers/deployment/).

What do you think?

**Comment by Philips-SSP** [comment]
Hi guys, I have a similar problem.

Yesterday, I launched a RADON workspace. Today I wanted to open it again, but it was responding very slowly, so I decided to stop and delete the workspace. Now, when I'm trying to create a new RADON workspace, it gives me the following log:

Pulling image "quay.io/eclipse/che-plugin-metadata-broker:v3.1.2"
Successfully pulled image "quay.io/eclipse/che-plugin-metadata-broker:v3.1.2"
Created container che-plugin-metadata-broker-v3-1-2
Started container che-plugin-metadata-broker-v3-1-2
Starting plugin metadata broker
List of plugins and editors to install
- radon/radon-menu/0.0.1 - A plugin to add RADON custom commands
- eclipse/che-machine-exec-plugin/7.12.0 - Che Plug-in with che-machine-exec service to provide creation terminal or tasks for Eclipse Che workspace containers.
- eclipse/che-theia/next - Eclipse Theia, get the latest release each day.
All plugin metadata has been successfully processed
Error: Failed to run the workspace: "Failure executing: POST at: https://10.96.0.1/api/v1/namespaces/che/services. Message: services "radon-gmt" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=services, name=radon-gmt, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=services "radon-gmt" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={})."

**Comment by miwurster** [comment]
I don't think that increasing the number of replicas will solve the problem. What we instead need is the ability to launch multiple pods and services I think. But atm I don't know how to achieve this with the `devfile` mechanics.

**Comment by miwurster** [comment]
> Hi guys, I have a similar problem.
> 
> Yesterday, I launched a RADON workspace and played around a bit in the GMT-Winery. Today I wanted to open it again, but it was responding very slowly, so I decided to stop and delete the workspace. Now, when I'm trying to create a new RADON workspace, it gives me the following log:
> 
> Pulling image "quay.io/eclipse/che-plugin-metadata-broker:v3.1.2"
> Successfully pulled image "quay.io/eclipse/che-plugin-metadata-broker:v3.1.2"
> Created container che-plugin-metadata-broker-v3-1-2
> Started container che-plugin-metadata-broker-v3-1-2
> Starting plugin metadata broker
> List of plugins and editors to install
> -- radon/radon-menu/0.0.1 - A plugin to add RADON custom commands
> -- eclipse/che-machine-exec-plugin/7.12.0 - Che Plug-in with che-machine-exec service to provide creation terminal or tasks for Eclipse Che workspace containers.
> -- eclipse/che-theia/next - Eclipse Theia, get the latest release each day.
> All plugin metadata has been successfully processed
> Error: Failed to run the workspace: "Failure executing: POST at: [10.96.0.1/api/v1/namespaces/che/services](https://10.96.0.1/api/v1/namespaces/che/services). Message: services "radon-gmt" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=services, name=radon-gmt, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=services "radon-gmt" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={})."

The issue here is that Eclipse Che does not clean up the old workspace correctly (there a still some left-over Kubernetes resources). You can delete them manually by issuing e.g. `kubectl delete pods,services -l app=winery -n <namespace>` or `kubectl delete service radon-gmt -n <namespace>` (you need to check to which namespace Eclipse Che assigns the devfile resources.

**Comment by Philips-SSP** [comment]
> > Hi guys, I have a similar problem.
...
> 
> The issue here is that Eclipse Che does not clean up the old workspace correctly (there are still some left-over Kubernetes resources). You can delete them manually by issuing e.g. `kubectl delete pods,services -l app=winery -n <namespace>` or `kubectl delete service radon-gmt -n <namespace>` (you need to check to which namespace Eclipse Che assigns the devfile resources).
> Alternatively, you can try to install the Kubernetes Dashbaord onto your cluster: https://github.com/kubernetes/dashboard#getting-started

Thanks @miwurster, that makes sense. I am very new to the RADON IDE, apologies for the perhaps obvious questions:

1. Where can I issue these commands? Somewhere in Eclipse Che?
2. Does the <namespace> variable refer to the workspace name?
3. If so, how do I find the <namespace> of a workspace that I have already deleted?

My problem was resolved this morning btw, I don't know if any of you guys fixed it for me or perhaps it was resolved automatically.

**Comment by sdagostini** [comment]
> > Hi guys, I have a similar problem.
> > Yesterday, I launched a RADON workspace and played around a bit in the GMT-Winery. Today I wanted to open it again, but it was responding very slowly, so I decided to stop and delete the workspace. Now, when I'm trying to create a new RADON workspace, it gives me the following log:
> > Pulling image "quay.io/eclipse/che-plugin-metadata-broker:v3.1.2"
> > Successfully pulled image "quay.io/eclipse/che-plugin-metadata-broker:v3.1.2"
> > Created container che-plugin-metadata-broker-v3-1-2
> > Started container che-plugin-metadata-broker-v3-1-2
> > Starting plugin metadata broker
> > List of plugins and editors to install
> > -- radon/radon-menu/0.0.1 - A plugin to add RADON custom commands
> > -- eclipse/che-machine-exec-plugin/7.12.0 - Che Plug-in with che-machine-exec service to provide creation terminal or tasks for Eclipse Che workspace containers.
> > -- eclipse/che-theia/next - Eclipse Theia, get the latest release each day.
> > All plugin metadata has been successfully processed
> > Error: Failed to run the workspace: "Failure executing: POST at: [10.96.0.1/api/v1/namespaces/che/services](https://10.96.0.1/api/v1/namespaces/che/services). Message: services "radon-gmt" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=services, name=radon-gmt, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=services "radon-gmt" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={})."
> 
> The issue here is that Eclipse Che does not clean up the old workspace correctly (there are still some left-over Kubernetes resources). You can delete them manually by issuing e.g. `kubectl delete pods,services -l app=winery -n <namespace>` or `kubectl delete service radon-gmt -n <namespace>` (you need to check to which namespace Eclipse Che assigns the devfile resources).
> Alternatively, you can try to install the Kubernetes Dashbaord onto your cluster: https://github.com/kubernetes/dashboard#getting-started

Dear @miwurster, I also would like to better understand where you apply this command. Are you considering a local instance of Eclipse Che? Because if we take into consideration the instance of Eclipse Che that is running on the ENG server (the one that  @Philips-SSP is using) probably this command must to be applied on the ENG server (so by an ENG person)

**Comment by sdagostini** [comment]
> > > Hi guys, I have a similar problem.
> > > ...
> > 
> > 
> > The issue here is that Eclipse Che does not clean up the old workspace correctly (there are still some left-over Kubernetes resources). You can delete them manually by issuing e.g. `kubectl delete pods,services -l app=winery -n <namespace>` or `kubectl delete service radon-gmt -n <namespace>` (you need to check to which namespace Eclipse Che assigns the devfile resources).
> > Alternatively, you can try to install the Kubernetes Dashbaord onto your cluster: https://github.com/kubernetes/dashboard#getting-started
> 
> Thanks @miwurster, that makes sense. I am very new to the RADON IDE, apologies for the perhaps obvious questions:
> 
> 1. Where can I issue these commands? Somewhere in Eclipse Che?
> 2. Does the  variable refer to the workspace name?
> 3. If so, how do I find the  of a workspace that I have already deleted?
> 
> My problem was resolved this morning btw, I don't know if any of you guys fixed it for me or perhaps it was resolved automatically.

Hi @Philips-SSP, The problem has not yet been resolved, at the moment if seem related to a problem in the Eclipse Che technology. You succeed to open/create a workspace today because probably there is no other users working on a RADON workspace.
@miwurster, as evidenced by @Philips-SSP the problem in some cases is resolved without the need to make the delete command, I think because simply there aren't other connected users to a RADON workspace. I experimented the same think during the demo with Philips: one user had opened a RADON workspace and I was unable to open mine; as soon as he closed the workspace I was able to open it, without the need to make the delete command.

**Comment by miwurster** [comment]
Opened an issue <https://github.com/eclipse/che/issues/17028>

**Comment by miwurster** [comment]
@sdagostini the commands I mentioned must be executed on the Kubernetes cluster where Eclipse Che is installed. So yes, this commands must be executed from some admin person. Most probably from you guys if @Philips-SSP uses our hosted Che instance.

**Comment by miwurster** [comment]
According to <https://github.com/eclipse/che/issues/17028#issuecomment-635858460> we should set the discoverability of the endpoints to 'false'. An updated devfile can be found here: https://github.com/eclipse/winery/blob/master/deploy/che/devfile.yml

@sdagostini could you pls update this respectively in the RADON registry? ... or is the registry somewhere available on GitHub so that I can propose the change via a PR?

Regarding the second issue (service is not properly clean-up), this is an issue in Eclipse Che and should be resolved in version 7.13. 

@sdagostini we should update our Che server.

**Comment by sdagostini** [comment]
> According to [eclipse/che#17028 (comment)](https://github.com/eclipse/che/issues/17028#issuecomment-635858460) we should set the discoverability of the endpoints to 'false'. An updated devfile can be found here: https://github.com/eclipse/winery/blob/master/deploy/che/devfile.yml
> 
> @sdagostini could you pls update this respectively in the RADON registry? ... or is the registry somewhere available on GitHub so that I can propose the change via a PR?
> 
> Regarding the second issue (service is not properly clean-up), this is an issue in Eclipse Che and should be resolved in version 7.13.
> 
> @sdagostini we should update our Che server.

@miwurster thank you Michael for the update of the devfile. I will updated this new devfile also in the RADON registry (because I have access to the ENG machine hosting Che)

I agree with you that we should update our Che server to version 7.13. My only doubt is that during the last update to a new version there were some problems and I would like to avoid that again. I will let you know.

**Comment by sdagostini** [comment]
The new devfile has been updated in the RADON registry
